### ISSUE TITLE
Update JSON Schemas for reconciled state machine

### ISSUE THREAD

**Issue by thekaveman** [openmobilityfoundation/mobility-data-specification#526]
After #506 is merged, we'll have a reconciled state machine between Agency and Provider. A follow-up task is to update the JSON Schemas with the new set of `vehicle_state` and `event_type`. 

If possible, we'd also like to indicate which combinations of those values are considered valid under the state machine as is done now within the [`status_changes.json`](https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/provider/dockless/status_changes.json#L324) Schema. This could be difficult because the new model allows an array of `event_type` per `vehicle_state`, and intermediate states are "hidden" by the final valid event transition. [See discussion on #271 for more details](https://github.com/openmobilityfoundation/mobility-data-specification/issues/271#issuecomment-624170186).

**Comment by thekaveman** [comment]
As expected, indicating the valid `vehicle_state` and `vehicle_events` combinations proved to be a challenge. 

In conversation with @karcass, we originally thought that the last entry in the `vehicle_events`  array should be a valid event transition for the indicated `vehicle_state`, e.g.

```json
{
    "vehicle_state": "non_operational",
    "event_types": ["trip_end", "battery_low"]
}
```

Here, the `battery_low` event is a valid transition into the `non_operational` state (while `trip_end` is not), and so it comes last in the array.

However, JSON Schema does not support the notion of checking the "last item in an array" - support is for checking a fixed number of items, or all items.

The solution I landed on was to simply check that the `event_types` array **contains** a valid event for the given `vehicle_state`. Thus, both the above example and the following would validate against the JSON Schema:

```json
{
    "vehicle_state": "non_operational",
    "event_types": ["battery_low", "trip_end"]
}
```

I think this is an OK trade-off in that we'll ensure that each `vehicle_state` has a correspondingly valid `event_type`, but we do lose the inherent "ordering" of the `event_types` array, from the perspective of the JSON Schemas anyway.

**Comment by schnuerle** [comment]
Thanks @thekaveman that looks like the best solution.

**Comment by thekaveman** [comment]
I just realized another approach: reverse the order, and have the first array item be the `event_type` that transitioned to the `vehicle_state`... so the events are in descending time order instead of ascending. We can revisit if needed.
